python/python/tests/test_dataset.py

```diff
@@ -898,6 +912,38 @@ def test_merge_insert(tmp_path: Path):
         dataset.merge_insert("a").execute(new_table)
 
 
+def test_merge_insert_conditional_upsert_example(tmp_path: Path):
```

Comment:
This has conditional update, but I wonder about what happens when there are also inserts? (`when_not_matched_insert_all`)

Reply:
There is a test for this in the rust unit tests (and I think in the python unit test above).  This unit test is actually superfluous but I wanted to make sure the exact scenario that the user was asking for worked.